content/how-to-use-supertokens-custom-ui-with-vuejs/index.md

```diff
@@ -0,0 +1,608 @@
+ },
+...
+```
+
+To load the `HomeView` component on `/` we’ll update the `/src/router/index.ts` file:
+
+```typescript
+...
+
+const router = createRouter({
+    history: createWebHistory(import.meta.env.BASE_URL),
+    routes: [
+        {
+            path: "/",
+            name: "home",
+            component: () => HomeView,
+        },
+        ...
+        
+    ],
+});
+
+export default router;
+
+```
+
+If you now visit [http://localhost:3000](http://localhost:3000/) after authentication,  you should see the following page:
+
+
+
+![homeview-after-login](./homeview_after_login.png)
+
+
+
+## Forgot Password
+
+In the Sign In UI, we have a link to the forgot password page. On this page the user can enter their email and receive a password reset link in their inbox. When they visit that link, they can then enter their new password on that page to reset their old password. 
+
+First we'll create a ForgotPassword HTML template inside` /src/html/forgotPassword.html`. An example of that can be found [here](https://github.com/supertokens/supertokens-web-js/blob/master/examples/vuejs/with-thirdpartyemailpassword/src/html/forgotPassword.html). 
+
+We'll create a component inside `/src/views/ForgotPassword.vue` file where we will render the above template:
+
+```vue
+<template src="../html/forgotPassword.html"></template>
+
+```
+
+In the HTML template above, we conditionally render a form based on a variable called `tokenPresent` (which is a state variable representing if a password reset token has been generated or not). This `tokenPresent` variable is set based on the token present in the query parameters of the page’s URL. In the case where the user has clicked on the forgot password button, there is no token present in the query parameters of the page’s URL, hence the `tokenPresent` variable is set to false. 
+
+Since  `tokenPresent` is false.  , we render the form where the user will enter their email to get the reset password link. When the user enters their email on this form and submits it, we call the `sendPasswordResetEmail` method from `supertokens-web-js` and pass their email inside it. This function interacts with a backend API to send a password reset link on the user’s email.
+
+The password reset link is of the form [http://localhost:3000/auth/reset-password?token=&rid=thirdpartyemailpassword]( http://localhost:3000/auth/reset-password?token=&rid=thirdpartyemailpassword). This link has the same path as the forgot password page, however, since the URL has the token query parameter, it will render the form where the user can enter their new password. 
+
+When they enter their new password in the form, we call the submitNewPassword method from supertokens-web-js with the new password. This function will read the token from the URL and call the backend API to change the user’s password.
+
+In case the token was consumed already or had expired, we show a message on the UI to prompt the user to go back to the login screen. 
```

Comment:
```suggestion
In case the token was consumed already or has expired, the function call will return a non `"OK"` status and then we can show a message on the UI to prompt the user to go back to the login screen. 
```